### PR TITLE
Add Product And Product Images

### PR DESCRIPTION
**Pull request by saadali5997**
1. Added product model with crud functionality and tests.
2. Added product_images model with crud functionality and tests.

### CODE REVIEW

File: product/models.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+            self.price,
+            self.stock
+        )
+
+
+class ProductImages(models.Model):
+    """Product image model."""
+
+    product = models.ForeignKey(Product, on_delete=models.CASCADE)
+    image1 = models.ImageField()
```
Comment (Ayub-Khan) [inline product/models.py]: https://docs.djangoproject.com/en/2.2/topics/files/

**Review by Ayub-Khan — CHANGES_REQUESTED** [review]

File: order/models.py [hunk 2]
```diff
@@ -4,13 +4,14 @@
...
 from employee.models import Employee
+from product.models import Product
 
 
 class Order(models.Model):
     """Order model."""
 
     client = models.ForeignKey(Client, on_delete=models.CASCADE)
-    product = models.PositiveIntegerField(blank=True, default=None, null=True)
+    product = models.ForeignKey(Product, blank=True, default=None, null=True, on_delete=models.SET_NULL)
```
Comment (Ayub-Khan) [inline order/models.py]: Why not remove when product is deleted.

File: product/models.py [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+from django.db import models
+from django.dispatch import receiver
+from django.utils.translation import ugettext_lazy as _
+
+
+class Product(models.Model):
+    """Product model."""
+
+    title = models.CharField(max_length=30, unique=True)
+    product_type = models.CharField(max_length=30)
```
Comment (Ayub-Khan) [inline product/models.py:17]: add a new filed product description.

File: product/models.py [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+            self.title,
+            self.price,
+            self.stock
+        )
+
+
+class ProductImages(models.Model):
+    """Product image model."""
+
+    product = models.OneToOneField(Product, on_delete=models.CASCADE)
```
Comment (Ayub-Khan) [inline product/models.py:37]: delete image files when deleting model field.

File: product/views/product_images_views.py [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+class ProductImagesListView(LoginRequiredMixin, View):
+    """List images against a product view."""
+
+    def get(self, request, product_id):
+        """Render product images list template.."""
+        product = get_object_or_404(Product, id=product_id)
+        product_images = get_object_or_404(ProductImages, product=product)
+        template = loader.get_template('product_images/list-product_images.html')
+        pi_dict = model_to_dict(product_images)
+        pi_dict.pop('id')
```
Comment (Ayub-Khan) [inline product/views/product_images_views.py]: use self-explanatory names.  product_images_dict or product_images.

File: product/views/product_images_views.py [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+        context = {
+            'product_images': pi_dict,
+            'product': product
+        }
+        return HttpResponse(template.render(context, request))
```
Comment (Ayub-Khan) [inline product/views/product_images_views.py]: why not use render here too ?

File: product/views/product_views.py [hunk 7]
```diff
@@ -0,0 +1,102 @@
...
+        return HttpResponse(template.render(context, request))
+
+
+product_detail_view = ProductDetailView.as_view()
+
+
+class ProductDeleteView(LoginRequiredMixin, View):
+    """Delete product view."""
+
+    def get(self, request, id):
```
Comment (Ayub-Khan) [inline product/views/product_views.py]: use post .

**Review by Ayub-Khan — CHANGES_REQUESTED** [review]

File: order/models.py (hunk 2 above)
Comment (saadali5997) [reply order/models.py]: > Why not remove when product is deleted.

If the product is deleted, that order becomes a custom order i.e. an order which is not a product.

File: product/views/product_views.py (hunk 7 above)
Comment (saadali5997) [reply product/views/product_views.py]: > use post .

@Ayub-Khan  all our `delete` views are `get` type views, will it be okay having this one alone as `post`?

Comment (Ayub-Khan) [reply product/views/product_views.py]: ah update all to post.

File: order/models.py (hunk 2 above)
Comment (Ayub-Khan) [reply order/models.py]: no need just delete the order.

**Review by Ayub-Khan — APPROVED** [review]

**Pull request merged**